### ISSUE TITLE
cpu.Info() returns error when clock speed not available

### ISSUE THREAD

**Issue by tmm1** [shirou/gopsutil#282]
I'm getting the following error from `cpu.Info()`:

```
open /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq: no such file or directory
```

on an ARM device that has no clock speed entry in`/proc/cpuinfo`:

```
$ cat /proc/cpuinfo
Processor	: Marvell PJ4Bv7 Processor rev 2 (v7l)
processor	: 0
BogoMIPS	: 1332.01

processor	: 1
BogoMIPS	: 1332.01

Features	: swp half thumb fastmult vfp edsp vfpv3 tls
CPU implementer	: 0x56
CPU architecture: 7
CPU variant	: 0x2
CPU part	: 0x584
CPU revision	: 2

Hardware	: Marvell Armada XP Development Board
Revision	: 0000
Serial	: 0000000000000000
```

and no `cpufreq` node available under `/sys/.../cpu`:

```
$ ls /sys/devices/system/cpu/cpu0/
cpuidle online topology
```

In this case, I think it would be useful to return the known information about the CPUs, even if the Mhz field is set to 0.

**Comment by tmm1** [comment]
This specific device is a Synology branded RS815 NAS. The admin web UI for the device shows a clock speed of 1.33ghz, so perhaps gopsutil should fallback to the `BogoMIPS` entry in `/proc/cpuinfo` when no other clock speed information is available?

**Comment by tmm1** [comment]
FYI I found another device running linux 3.2.26 where `/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq` is reported in Hz instead of KHz.

**Comment by schmichael** [comment]
From the little bit of work I've done on ARM it seems that missing cpufreq is common and likely never going to be fully covered.

So on systems lacking cpufreq it might be worth falling back to `dmidecode -s processor-frequency` which works in the two test cases I have available.

There are 3 significant downsides to this:

1. dmidecode is rarely if ever installed by default on systems
2. dmidecode seems to require root (or at least read access to pseudo-files that are only readable by root on my system)
3. dmidecode's man page has a big warning about how untrustworthy it is:

> More often than not, information contained in the DMI tables is inaccurate, incomplete or simply wrong.

Still seems worth trying in a fallback.

**Comment by shirou** [comment]
@tmm1 Sorry for very very late response. I created #335 which just ignore error. Does this resolve your problem?

@schmichael  Thank you for researching. I agree your downside view about `dmidecode`, especially it needs `root`.

**Comment by schmichael** [comment]
@shirou Seems reasonable to me. Could I open a new issue (and maybe a PR someday soon) for adding a dmidecode fallback? I know it's of extremely limited usefulness, so feel free to say "no thanks."

**Comment by tmm1** [comment]
@shirou Thanks, that resolves my issue. I also have some other changes locally which you might find useful:

fancybits/gopsutil@8abfc8b8c5202a08ac6f47ce3ad6caabb4c60650 parse bogomips values
fancybits/gopsutil@d3bf1ea7b3713861e487c1b330b8bcf1f7ee9e12 use processor name as model fallback
fancybits/gopsutil@70e6762eb9d1aa521d43ed2bf4a3863c2c41a578  handle older kernels which list cpu speed in hz

@schmichael Thanks for the `dmidecode` tip. Unfortunately none of the ARM systems which I have seen have that tool installed, and most don't allow me to run things as root. So this particular workaround would not be very useful to me, atleast.

**Comment by shirou** [comment]
@tmm1 sorry, when I close #335, also closed this issue.
And thank you for your commit. according to your valuable commits, I did two commit. I didn't include `parse bogomips values` commit, because `Bogomips` is too linux specific.

3499dec5c013d054d5d2a99ec48345998034e841
a63f1aeeecf449baad9cbb2ee2b6e252accdbf5e
I have wanted to make a PR in order to note your name, but I failed push branch. very sad. sorry. But I really thank you for these.
